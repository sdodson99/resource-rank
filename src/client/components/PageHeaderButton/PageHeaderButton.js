import React from 'react';
import PropTypes from 'prop-types';
import styles from './PageHeaderButton.module.css';
import Link from 'next/link';

const PageHeaderButton = ({ title, buttonContent, hideButton, linkTo }) => (
  <div className={styles.PageHeaderButton} data-testid="PageHeaderButton">
    <div className="flex items-center justify-between text-center sm:text-left">
      <div className="text-4xl">{title}</div>

      {hideButton && (
        <div className="ml-3 flex">
          <Link href={linkTo}>
            <a className="btn btn-primary">{buttonContent}</a>
          </Link>
        </div>
      )}
    </div>
  </div>
);

PageHeaderButton.propTypes = {
  title: PropTypes.string,
  buttonContent: PropTypes.string,
  hideButton: PropTypes.bool,
  linkTo: PropTypes.string,
};

PageHeaderButton.defaultProps = {
  linkTo: '',
};

export default PageHeaderButton;
