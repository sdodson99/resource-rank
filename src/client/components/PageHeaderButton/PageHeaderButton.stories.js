import React from 'react';
import PageHeaderButton from './PageHeaderButton';

export default {
  component: PageHeaderButton,
  title: 'Components/PageHeaderButton',
  decorators: [
    (Story) => {
      return <Story />;
    },
  ],
};

const Template = (args) => <PageHeaderButton {...args} />;

export const Default = Template.bind({});
Default.args = {
  title: 'Page Title',
  buttonContent: 'New',
};

export const HideButton = Template.bind({});
HideButton.args = {
  ...Default.args,
  hideButton: true,
};

export const TitleStyle = Template.bind({});
TitleStyle.args = {
  ...Default.args,
  titleClassName: 'text-xl text-blue-500',
};
