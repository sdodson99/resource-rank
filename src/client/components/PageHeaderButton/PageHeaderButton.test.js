import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import PageHeaderButton from './PageHeaderButton';
import renderer from 'react-test-renderer';

describe('<PageHeaderButton />', () => {
  test('it should mount', () => {
    render(<PageHeaderButton />);

    const pageHeaderButton = screen.getByTestId('PageHeaderButton');

    expect(pageHeaderButton).toBeInTheDocument();
  });

  it('should render correctly', () => {
    const tree = renderer
      .create(<PageHeaderButton titleClassName="text-3xl" />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  describe('with button hidden', () => {
    it('should render correctly', () => {
      const tree = renderer
        .create(<PageHeaderButton hideButton={true} />)
        .toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
