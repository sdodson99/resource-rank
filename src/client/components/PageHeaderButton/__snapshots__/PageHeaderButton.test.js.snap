// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageHeaderButton /> should render correctly 1`] = `
<div
  className="PageHeaderButton"
  data-testid="PageHeaderButton"
>
  <div
    className="flex items-center justify-between"
  >
    <div
      className="text-4xl"
    />
  </div>
</div>
`;

exports[`<PageHeaderButton /> with button hidden should render correctly 1`] = `
<div
  className="PageHeaderButton"
  data-testid="PageHeaderButton"
>
  <div
    className="flex items-center justify-between"
  >
    <div
      className="text-4xl"
    />
    <div>
      <a
        className="btn btn-primary"
        href="/"
        onClick={[Function]}
        onMouseEnter={[Function]}
      />
    </div>
  </div>
</div>
`;
